src/classes/groundplaces.ts

```diff
@@ -25,21 +48,95 @@ export class GroundPlaces {
   public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete {}
 
   /**
-   * @description Create the stopGroup with the values given.
+   * @description Create a new StopGroup from a SegmentProviderStop.
+   * @param {SegmentProviderStop} segmentProviderStop - The SegmentProviderStop on which is based the StopGroup.
+   * @param {StopGroupInfos} stopGroupInfos - StopGroup informations.
    * @param {StopClusterGpuid} stopClusterGpuid - Ground place unique identifier of the stopCluster parent.
-   * @param {StopGroup} stopGroupInfos - StopGroup informations.
-   * @returns {StopGroup}
+   * @returns {StopGroup|Error}
    */
-  // @ts-ignore
-  public createStopGroup(stopClusterGpuid: StopClusterGpuid, stopGroupInfos: StopGroup): void {}
+  public createStopGroup(
+    segmentProviderStop: SegmentProviderStop,
+    stopGroupInfos: StopGroupInfos,
+    stopClusterGpuid: StopClusterGpuid,
+    // @ts-ignore
+  ): StopGroup | Error {
+    const { id: stopGroupGpuid }: GroundPlaceGenerated = this.generatorService.gpuid(stopGroupInfos);
+    const { latitude, longitude, name, countryCode, currentStopGroupGpuid, serviced } = stopGroupInfos;
+    const { id: segmentProviderStopId } = segmentProviderStop;
+
+    const createStopGroupAction = {
+      [stopGroupGpuid]: {
+        latitude,
+        longitude,
+        name,
+        country_code: countryCode,
+        childs: [segmentProviderStop],
+        serviced,
+        type: Create,
+      },
+    };
+
+    const moveStopGroupToStopClusterAction = {
+      [stopGroupGpuid]: {
+        into: stopClusterGpuid,
+        type: Move,
+      },
+    };
+
+    const moveSegmentProviderStopAction = {
+      [currentStopGroupGpuid]: {
+        segmentProviderStopId,
+        into: stopClusterGpuid,
+        type: MoveSegmentProviderStop,
+      },
+    };
+
+    this.addGroundPlacesDiffActions([
+      createStopGroupAction,
+      moveStopGroupToStopClusterAction,
+      moveSegmentProviderStopAction,
+    ]);
+
+    // TODO: Call applyGroundPlacesDiff and getStopGroup
+    // this.applyGroundPlacesDiff();
```

Comment:
Wow ! OK ! I just get what you got wrong !
Your method is SUPPOSED to do the action on you storage and so on.
THEN log the action in the GroundPlacesDiff.

------- 
On the backend LATER you'll just have to APPLY the ground places diff and it will executes the same function.

Reply:
You will apply the DIFF only on the backend side ! And not every action !